tests/helpers/kube.go

```diff
@@ -610,3 +611,51 @@ func CreateRebootNodeCR(
 
 	return rebootNode, nil
 }
+
+// WaitForNodeConditionWithCheckName waits for the node to have a condition with the reason containing the specified checkName.
+func WaitForNodeConditionWithCheckName(ctx context.Context, t *testing.T, c klient.Client, nodeName, checkName string) {
+	require.Eventually(t, func() bool {
+		node, err := GetNodeByName(ctx, c, nodeName)
+		if err != nil {
+			t.Logf("failed to get node %s: %v", nodeName, err)
+			return false
+		}
+
+		// Look for a condition where the reason contains the check name
+		for _, condition := range node.Status.Conditions {
+			if condition.Status == v1.ConditionTrue && strings.Contains(condition.Reason, checkName) {
```

Comment:
Can we make this more strict -- validate all the fields -- type, status, reason and message and not just a subset?